contracts/vesting/src/storage.rs

```diff
@@ -0,0 +1,155 @@
+use curve::Curve;
+use soroban_sdk::{
+    contracttype, log, panic_with_error, Address, ConversionError, Env, String, TryFromVal, Val,
+    Vec,
+};
+
+use crate::error::ContractError;
+
+impl TryFromVal<Env, DataKey> for Val {
+    type Error = ConversionError;
+
+    fn try_from_val(_env: &Env, v: &DataKey) -> Result<Self, Self::Error> {
+        Ok((*v as u32).into())
+    }
+}
+
+#[derive(Clone, Copy)]
+#[repr(u32)]
+pub enum DataKey {
+    Admin = 1,
+    Config = 2,
+    Minter = 3,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Config {
+    /// `admin` who can manage the contract with administrative privileges.
+    pub admin: Address,
```

Comment:
We already have an admin as a datakey, why putting him in the Config again? 
I would even consider removing config and saving each seaparately - I noticed most of the time only one parameter is necessary from the whole config. 
If it's not optimizing anything, just spread it.

Reply:
Config completely removed